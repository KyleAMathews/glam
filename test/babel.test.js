/* eslint-disable no-useless-escape */
/* eslint-env jest */
const path = require('path');
const pluginTester = require('babel-plugin-tester')
const glamPlugin = require('../src/babel')

pluginTester({
  plugin: glamPlugin,
  snapshot: true,
  fixtures: path.join(__dirname, '__fixtures__'),
  tests: [
    `
      css\`
       margin: 12px 48px;
       color: #ffffff;
       height: \$\{props => props.height * props.scale\};
       display: flex;
       flex: 1 0 auto;
       color: blue;
       width: \$\{widthVar\};
    \``,
    `
      const frag = fragment\`padding: 8px;\`
      const fragB = fragment\`height: \$\{heightVar\}\`
      css\`
       @apply \$\{frag\};
       @apply $\{fragB\}; 
       margin: 12px 48px;
       color: #ffffff;
    \``
  ]
})
