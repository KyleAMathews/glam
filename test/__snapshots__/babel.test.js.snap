// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. glam 1`] = `
"
css\`
   margin: 12px 48px;
   color: #ffffff;
   height: \${props => props.height * props.scale};
   display: flex;
   flex: 1 0 auto;
   color: blue;
   width: \${widthVar};
\`

      ↓ ↓ ↓ ↓ ↓ ↓

css(\\"css-oxz0pg\\", [props => props.height * props.scale, widthVar]);
"
`;

exports[`2. glam 1`] = `
"
const frag = fragment\`padding: 8px;\`
  const fragB = fragment\`height: \${heightVar}\`
  css\`
   @apply \${frag};
   @apply \${fragB}; 
   margin: 12px 48px;
   color: #ffffff;
\`

      ↓ ↓ ↓ ↓ ↓ ↓

const frag = fragment('frag-zzm2ov');
const fragB = fragment('frag-8zv2nf', [heightVar]);
css(\\"css-t2fxs5\\", [frag, fragB]);
"
`;
